docker/docker_image_dest.go

```diff
@@ -781,9 +813,20 @@ func (d *dockerImageDestination) putSignaturesToSigstoreAttachments(ctx context.
 	if err != nil {
 		return err
 	}
-	ociManifest.Config = configDesc
 
-	manifestBlob, err := ociManifest.Serialize()
+	switch v := genManifest.(type) {
+	case *manifest.OCI1:
+		v.Config = configDesc
+	case *manifest.Schema2:
+		v.ConfigDescriptor = manifest.Schema2Descriptor{
```

Comment:
- The config was uploaded above with a hard-coded MIME type
- Structurally, assuming that OCI and schema2 configs are completely substitutable is not something the transport should assume; such a no-op conversion belongs into the `image/` conversion code.